File: src/main/java/nl/rabobank/kotlinmovement/recipes/domain/IngredientRequest.java
package nl.rabobank.kotlinmovement.recipes.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class IngredientRequest {
    @NotBlank
    private String name;
    @NotBlank
    private IngredientType type;
    @NotBlank
    private int weight;
}

File: src/test/kotlin/nl/rabobank/kotlinmovement/recipes/testutil/testdata/IngredientRequest.kt
package nl.rabobank.kotlinmovement.recipes.testutil.testdata

import javax.validation.constraints.NotBlank

class IngredientRequest (
    val name: @NotBlank String? = null,
    val type: @NotBlank IngredientType? = null,
    val weight: Int = 0
)
